openassessment/xblock/staff_area_mixin.py

```diff
@@ -0,0 +1,391 @@
+from functools import wraps
+import logging
+
+from xblock.core import XBlock
+from openassessment.assessment.errors import (
+    PeerAssessmentInternalError,
+)
+from openassessment.workflow.errors import (
+    AssessmentWorkflowError, AssessmentWorkflowInternalError
+)
+from openassessment.assessment.errors.ai import AIError
+from openassessment.xblock.resolve_dates import DISTANT_PAST, DISTANT_FUTURE
+from openassessment.xblock.data_conversion import (
+    create_rubric_dict, convert_training_examples_list_to_dict, create_submission_dict
+)
+from submissions import api as submission_api
+from openassessment.assessment.api import peer as peer_api
+from openassessment.assessment.api import self as self_api
+from openassessment.assessment.api import ai as ai_api
+from openassessment.fileupload import api as file_api
+from openassessment.workflow import api as workflow_api
+from openassessment.fileupload import exceptions as file_exceptions
+
+
+logger = logging.getLogger(__name__)
+
+
+def require_global_admin(error_key):
+    """
+    Method decorator to restrict access to an XBlock handler
+    to only global staff.
+
+    Args:
+        error_key (str): The key to the error message to display to the user
+        if they do not have sufficient permissions.
+
+    Returns:
+        Decorated function
+
+    """
+    def _decorator(func):  # pylint: disable=C0111
+        @wraps(func)
+        def _wrapped(xblock, *args, **kwargs):  # pylint: disable=C0111
+            permission_errors = {
+                "SCHEDULE_TRAINING": xblock._(u"You do not have permission to schedule training"),
+                "RESCHEDULE_TASKS": xblock._(u"You do not have permission to reschedule tasks."),
+            }
+            if not xblock.is_admin or xblock.in_studio_preview:
+                return {'success': False, 'msg': permission_errors[error_key]}
+            else:
+                return func(xblock, *args, **kwargs)
+        return _wrapped
+    return _decorator
+
+
+def require_course_staff(error_key, with_json_handler=False):
+    """
+    Method decorator to restrict access to an XBlock render
+    method to only course staff.
+
+    Args:
+        error_key (str): The key for the error message to display to the
+            user if they do not have sufficient permissions.
+
+    Returns:
+        decorated function
+
+    """
+    def _decorator(func):  # pylint: disable=C0111
+        @wraps(func)
+        def _wrapped(xblock, *args, **kwargs):  # pylint: disable=C0111
+            permission_errors = {
+                "STAFF_AREA": xblock._(u"You do not have permission to access staff information"),
+                "STUDENT_INFO": xblock._(u"You do not have permission to access learner information."),
+            }
+
+            if not xblock.is_course_staff and with_json_handler:
+                return {"success": False, "msg": permission_errors[error_key]}
+            elif not xblock.is_course_staff or xblock.in_studio_preview:
```

Comment:
That's an interesting question. I didn't introduce this code (GitHub is doing a poor job of understanding renames) so I'd prefer to leave it as is. @explorerleslie, any thoughts as to whether the staff area should be shown in Studio preview?